mailbox/api/src/test/java/org/apache/james/mailbox/events/GroupContract.java

```diff
@@ -96,6 +100,74 @@ public void event(Event event) throws Exception {
             assertThat(rateExceeded).isFalse();
         }
 
+        @Test
+        default void groupNotificationShouldDeliverASingleEventToAllListenersAtTheSameTime() {
+            CountDownLatch countDownLatch = new CountDownLatch(1);
+            try {
+                ConcurrentLinkedQueue<String> threads = new ConcurrentLinkedQueue<>();
+                eventBus().register(new MailboxListener.GroupMailboxListener() {
+                    @Override
+                    public Group getDefaultGroup() {
+                        return new GenericGroup("group");
+                    }
+
+                    @Override
+                    public boolean isHandling(Event event) {
+                        return true;
+                    }
+
+                    @Override
+                    public void event(Event event) throws Exception {
+                        threads.add(Thread.currentThread().getName());
+                        countDownLatch.await();
+                    }
+                }, GROUP_A);
+                eventBus().register(new MailboxListener.GroupMailboxListener() {
+                    @Override
+                    public Group getDefaultGroup() {
+                        return new GenericGroup("groupB");
+                    }
+
+                    @Override
+                    public boolean isHandling(Event event) {
+                        return true;
+                    }
+
+                    @Override
+                    public void event(Event event) throws Exception {
+                        threads.add(Thread.currentThread().getName());
+                        countDownLatch.await();
+                    }
+                }, GROUP_B);
+                eventBus().register(new MailboxListener.GroupMailboxListener() {
+                    @Override
+                    public Group getDefaultGroup() {
+                        return new GenericGroup("groupC");
+                    }
+
+                    @Override
+                    public boolean isHandling(Event event) {
+                        return true;
+                    }
+
+                    @Override
+                    public void event(Event event) throws Exception {
+                        threads.add(Thread.currentThread().getName());
+                        countDownLatch.await();
+                    }
+                }, GROUP_C);
+
+                eventBus().dispatch(EVENT, NO_KEYS).subscribeOn(Schedulers.elastic()).subscribe();
+
+
+                WAIT_CONDITION.atMost(org.awaitility.Duration.TEN_SECONDS)
+                    .untilAsserted(() -> assertThat(threads.size()).isEqualTo(3));
```

Comment:
Should the threads be distinct?

Please use `hasSize` method from assertj

Reply:
> Should the threads be distinct?

Of course, it's next line assertion, the most important one.